src/Tuner.cpp

```diff
@@ -382,18 +378,25 @@ std::vector<Parameters> Tuner<net_t>::build_valid_params() {
             Parameters param = get_parameters_by_int(opts, i);
             param["TCE"] = tce;
             if (valid_config_sgemm(param, cfg_sgemm_exhaustive)) {
-                if (cfg_sgemm_exhaustive) {
-                    if (rng.randfix<16>() != 0) {
-                        continue;
-                    }
-                }
                 valid_params.push_back(param);
             }
         }
     };
     build_from(opts, 0);
     build_from(topts, 1);
 
+    // Don't use thread RNG or determinism will depend on whether tuner ran.
+    auto rng = std::default_random_engine{};
```

Comment:
Please use the included Rng. It will be better and faster than the system one in almost all cases (blame C++11).

(The comment would also make no sense otherwise)